integration-libs/opf/base/core/services/opf-cart-handler.service.ts

```diff
@@ -320,31 +338,30 @@ export class OpfCartHandlerService {
   loadCartAfterSingleProductTransaction(
     transactionDetails: QuickBuyTransactionDetails,
     orderSuccess = false
-  ): void {
+  ): Observable<boolean> {
```

Comment:
we don't subscribe anymore to avoid race conditions.
It allows to make sure 'deleteStaleCart()' as been completed (via eventService) before making any other cleaning operations like 'delete user associated addresses'